base/math.jl

```diff
@@ -629,9 +629,17 @@ julia> hypot(3, 4im)
 hypot(x::Number, y::Number) = hypot(promote(x, y)...)
 hypot(x::Complex, y::Complex) = hypot(abs(x), abs(y))
 hypot(x::T, y::T) where {T<:Real} = hypot(float(x), float(y))
-hypot(x::T, y::T) where {T<:Number} = (z = y/x; abs(x) * sqrt(one(z) + z*z))
+function hypot(x::T, y::T) where {T<:Number}
+    # Return Inf if either or both inputs is Inf (Compliance with IEEE754)
+    if isinf(x) || isinf(y)
```

Comment:
This method should not be called for IEEE754 floating point numbers; that case is caught by more specific methods. Putting float-specific behaviors here defeats the purpose of this, which is to handle very general number-like types that might not support all operations.

Reply:
Which is why I originally proposed `sqrt(abs2(x) + abs2(y))`, which is what you get if you pass 3 arguments to `hypot`. I'd be fine with that too.

Reply:
I see. How should I update the tests? Should I just delete the tests with `NaN` and `Inf`?

Reply:
Yes I think so. We'll still be following IEEE754 if you pass actual Float values, plus I think the specified behavior is really bizarre and unexpected. We already broke with it for `min` and `max` on `NaN`.